Symfony/tests/AppBundle/Entity/TierDatabaseTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Tests\AppBundle\Entity;
+
+use AppBundle\Entity\Tier;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+class TierDatabaseTest extends KernelTestCase
+{
+    /**
+     * @var \Doctrine\ORM\EntityManager
+     */
+    private $em;
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp()
+    {
+        self::bootKernel();
+
+        $this->em = static::$kernel->getContainer()
+            ->get('doctrine')
+            ->getManager();
+    }
+
+    protected function clearDatabase()
+    {
+        $tiers = $this->em
+            ->getRepository('AppBundle:Tier')
+            ->findAll();
+
+        foreach ($tiers as $tier) {
+            $this->em->remove($tier);
+        }
+        $this->em->flush();
+    }
+
+    public function testEmptyDatabase()
+    {
+        $this->clearDatabase();
+
+        $tiers = $this->em
+            ->getRepository('AppBundle:Tier')
+            ->findAll();
+
+        $this->assertCount(0, $tiers);
+    }
+
+    public function testNonEmptyDatabase()
+    {
+        $this->clearDatabase();
+
+        $tier = new Tier();
+        $tier->setTierName('A Foo Bar');
+        $tier->setPrice(19.99);
+        $tier->setDescription('Lorem');
+        $tier->setTierImage('img/test');
+        $tier->setSuccessfulFlashCount(1);
+
+        $this->em->persist($tier);
+        $this->em->flush();
+
+        $tiers = $this->em
+            ->getRepository('AppBundle:Tier')
+            ->findAll();
+
+        $this->assertCount(1, $tiers);
```

Comment:
Sure thing! I thought it would be good to have a step-by-step process, number of tiers, followed by what is in the tier, etc. Would you like the first assert removed?